File: JavaScript/413-Arithmetic Slices.js
/**
 * @param {number[]} A
 * @return {number}
 */
var numberOfArithmeticSlices = function(A) {
    if (A.length < 3) return 0;
    let dp = new Array(A.length).fill(0);
    let res = 0;
    
    for (let i = 2; i<A.length; i++){
        if (A[i-1] - A[i-2] === A[i] - A[i-1]){
            dp[i] = dp[i-1] + 1;
            res += dp[i];
        }
    }
    return res;
};

File: 413-Arithmetic Slices.py
class Solution(object):
    def numberOfArithmeticSlices(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        if len(A) < 3:
            return 0
        
        res = 0
        dp = [0]*len(A)
        
        for i in range(2, len(A)):
            if A[i-1] - A[i-2] == A[i] - A[i-1]:
                dp[i] = dp[i-1] + 1
                res += dp[i]
        
        return res